sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2025 @@
+    /// TrueType - FIXED 1.0
+    _TrueType = 0x00010000,
+    /// WOFF 1.0 - 'wOFF'
+    Woff = 0x774f4646,
+    /// WOFF 2.0 - 'wOF2'
+    _Woff2 = 0x774f4632,
+}
+
+/// Used to attempt conversion from u32 to a Magic value.
+impl TryFrom<u32> for Magic {
+    type Error = ();
+
+    /// Tries to convert from u32 to a valid font version.
+    fn try_from(v: u32) -> core::result::Result<Self, Self::Error> {
+        match v {
+            //x if x == Magic::TrueType as u32 => Ok(Magic::TrueType),
+            //x if x == Magic::OpenType as u32 => Ok(Magic::OpenType),
+            x if x == Magic::Woff as u32 => Ok(Magic::Woff),
+            //x if x == Magic::Woff2 as u32 => Ok(Magic::Woff2),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Minimal byte-flopper for moving font header structures and known table
+/// contents in and out of storage.
+trait NetworkByteOrderable {
+    fn from_network(&mut self);
+    fn to_network(&mut self);
+}
+
+/// 'C2PA' font table - in storage
+#[derive(Debug)]
+#[repr(C, packed)]
+#[allow(non_snake_case)]
+struct TableC2PARaw {
+    majorVersion: u16,
+    minorVersion: u16,
+    activeManifestUriOffset: u32,
+    activeManifestUriLength: u16,
+    manifestStoreOffset: u32,
+    manifestStoreLength: u32,
+}
+
+impl NetworkByteOrderable for TableC2PARaw {
+    fn from_network(&mut self) {
+        self.majorVersion = u16::from_be(self.majorVersion);
+        self.minorVersion = u16::from_be(self.minorVersion);
+        self.activeManifestUriOffset = u32::from_be(self.activeManifestUriOffset);
+        self.activeManifestUriLength = u16::from_be(self.activeManifestUriLength);
+        self.manifestStoreOffset = u32::from_be(self.manifestStoreOffset);
+        self.manifestStoreLength = u32::from_be(self.manifestStoreLength);
+    }
+
+    fn to_network(&mut self) {
+        self.majorVersion = u16::to_be(self.majorVersion);
+        self.minorVersion = u16::to_be(self.minorVersion);
+        self.activeManifestUriOffset = u32::to_be(self.activeManifestUriOffset);
+        self.activeManifestUriLength = u16::to_be(self.activeManifestUriLength);
+        self.manifestStoreOffset = u32::to_be(self.manifestStoreOffset);
+        self.manifestStoreLength = u32::to_be(self.manifestStoreLength);
+    }
+}
+
+/// 'C2PA' font table - in storage
+#[derive(Clone)]
+#[allow(non_snake_case)]
+pub struct TableC2PA {
+    /// Major version of the C2PA table record
+    pub majorVersion: u16,
+    /// Minor version of the C2PA table record
+    pub minorVersion: u16,
+    /// Optional URI to an active manifest
+    pub activeManifestUri: Option<String>,
+    /// Optional embedded manifest store
+    manifestStore: Option<Box<[u8]>>,
+}
+
+impl TableC2PA {
+    /// Creates a new C2PA record with the current default version information.
+    pub fn new(active_manifest_uri: Option<String>, manifest_store: Option<Vec<u8>>) -> Self {
+        let boxed_data = manifest_store.map(|d| d.into_boxed_slice());
```

Comment:
Previous comment will clean this function up as well.

Reply:
it did, just as you foresaw.